src/classes/speciesAtom.cpp

```diff
@@ -158,6 +158,38 @@ OptionalReferenceWrapper<SpeciesBond> SpeciesAtom::getBond(const SpeciesAtom *pa
     return *result;
 }
 
+// Return pairs of atoms bound to this one which form suitable angles
+std::vector<std::pair<const SpeciesAtom *, const SpeciesAtom *>> SpeciesAtom::getAnglePairs(double maxAngle,
+                                                                                            bool uniquePairsOnly) const
+{
+    if (bonds_.size() < 2)
+        return {};
+
+    std::vector<std::pair<const SpeciesAtom *, const SpeciesAtom *>> pairs;
+
+    for (auto b1 = 0; b1 < bonds_.size() - 1; ++b1)
+    {
+        auto *i = bonds_[b1].get().partner(this);
+        for (auto b2 = b1 + 1; b2 < bonds_.size(); ++b2)
+        {
+            auto *k = bonds_[b2].get().partner(this);
+            if (i == k)
+                continue;
+
+            // Determine angle
+            auto angle = Box::literalAngleInDegrees(i->r(), r_, k->r());
+            if (angle < maxAngle)
+            {
+                pairs.emplace_back(i, k);
+                if (!uniquePairsOnly)
+                    pairs.emplace_back(k, i);
+            }
+        }
+    }
```

Comment:
```suggestion
    dissolve::for_each_pair(ParallelPolicies:seq, bonds_.begin(), bonds_.end(), [this, uniquePairsOnly, &pairs](int b1, const auto bond1, int b2, const auto bond2) {
      auto *i = bond1.get().partner(this);
      auto *k = bond2.get().partner(this);
      if (i == k)
        return;

      // Determine angle
      auto angle = Box::literalAngleInDegrees(i->r(), r_, k->r());
      if (angle < maxAngle)
      {
          pairs.emplace_back(i, k);
          if (!uniquePairsOnly)
              pairs.emplace_back(k, i);
      }
    });
```
I thought that we might be able to simplify things with our pair algorithm. Please note that I haven't tested this code!

Reply:
This looks nice and I should have done it this way before, but you've actually highlighted a bit of now-redundant code! This was previously used to get pairs of atoms to form rotation bases, but since we're not doing things geometrically any more I can just remove the function entirely.